stacks-common/src/deps_common/httparse/mod.rs

```diff
@@ -645,7 +647,7 @@ fn parse_reason<'a>(bytes: &mut Bytes<'a>) -> Result<&'a str> {
         } else if b == b'\n' {
             let res = str::from_utf8(bytes.slice_skip(1)).map_err(|_e| Error::Status)?;
             return Ok(Status::Complete(res));
-        } else if !((b >= 0x20 && b <= 0x7E) || b == b'\t') {
+        } else if !((0x20..=0x7E).contains(&b) || b == b'\t') {
```

Comment:
Can you check that this doesn't do an _O(n)_ search through `0x20..=0x7E`?

Reply:
This is just a  check if a byte b is within the RANGE of ASCII characters from 0x20 to 0x7E. This operation is O(1).